### ISSUE TITLE
commenting out cells, or choosing their language

### ISSUE THREAD

**Issue by gordonwoodhull** [att/rcloud#229]
A user requests:

> Could there be a way to comment-out or inactivate an entire cell by a single click?
> One click could make it active again, later.
> It would be easy as the delete icon, but allow you to get the code back very easily.  
> Another possibility would be to have a cell type that is easy to change. regular/markdown/comment are 3 possibilities.

**Comment by holyguster** [comment]
I was asked about this one again today; just making a couple of comments.

If we use .txt or .text, then we lose the original language, and also probably nice language-dependent features like automatic ACE highlighting and autocompletion.  Left-field suggestion.... how about overloading the extension and using something like .txtR, .txtpy, .txtmd, etc, so that the original language is maintained?

Any .txt* extension cell would not be executed at run time.  We would also need to very clearly indicate cells that are "commented out" in the UI, possibly a different color at edit time and some special message at run time like "cells of type comment are not executed".

**Comment by holyguster** [comment]
More thoughts... instead of having the user enable/disable this via the UI in the language pulldown, maybe there should be something like a "toggle comment" or "toggle ignore cell" button that turns this ignore cell feature on/off.  The icon for that button could be a pound sign.

**Comment by gordonwoodhull** [comment]
Yes, I came to the same conclusion as well. Although reusing the language select for this is convenient from a programming perspective, and avoids yet another button, really it's independent from the user's perspective.

Now the question is where to store the metadata. It's unfortunate that gists don't have an area for extra keys/values. I guess we'll have to keep the commentedness in RCS, which means it will get lost when the notebook is exported.

**Comment by holyguster** [comment]
Is overloading the extension, like renaming part1.R to part1.txtR to turn on "ignore", then renaming back to turn off "ignore", a PITA for storing the metadata?  Seems like that would be maintained on export, but given that I do not know the RCloud internals, I have no idea how much pain that would be.
